app/scripts/lib/transaction/eip5792.ts

```diff
@@ -0,0 +1,66 @@
+import { SendCalls } from '@metamask/eth-json-rpc-middleware/dist/methods/wallet-send-calls';
+import { NetworkController } from '@metamask/network-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { TransactionController } from '@metamask/transaction-controller';
+import { Hex, JsonRpcRequest } from '@metamask/utils';
+
+export async function processSendCalls(
+  transactionController: TransactionController,
+  networkController: NetworkController,
+  params: SendCalls,
+  req: JsonRpcRequest & { networkClientId: string; origin?: string },
+) {
+  const { calls, chainId: requestChainId, from } = params;
+  const { networkClientId, origin } = req;
+  const transactions = calls.map((call) => ({ params: call }));
+
+  const dappChainId =
+    networkController.getNetworkClientById(networkClientId).configuration
+      .chainId;
+
+  if (
+    requestChainId &&
+    requestChainId.toLowerCase() !== dappChainId.toLowerCase()
+  ) {
+    throw rpcErrors.invalidInput(
+      `Chain ID must match the dApp selected network: Got ${requestChainId}, expected ${dappChainId}`,
+    );
```

Comment:
Hmmmm I'm not sure if this is the right rpc error... Though I'm looking at the others and this may be our best match 🤔

Reply:
This is our standard bad input error we use for all the transaction and signature calls from `eth-json-rpc-middleware`.